Failed to initialize x86_energy metrics

I'm getting an error:

```
Failed to initialize x86_energy metrics: Error in function setup at /home/s9242987/Software/x86_energy/src/access/sysfs.c:181: could not get a file pointer to "/sys//devices/system/cpu/cpu31/cache/index1/shared_cpu_list"
```

File is readable for the user and there are file descriptors available.

Build error:

```
/home/s9242987/Software/x86_energy/src/access/sysfs.c: In function ‘setup’:
/home/s9242987/Software/x86_energy/src/access/sysfs.c:96:13: error: ‘totalfiles’ undeclared (first use in this function); did you mean ‘total_files’?
         if (totalfiles == 0)
             ^~~~~~~~~~
             total_files
/home/s9242987/Software/x86_energy/src/access/sysfs.c:96:13: note: each undeclared identifier is reported only once for each function it appears in
CMakeFiles/x86_energy-static.dir/build.make:278: recipe for target 'CMakeFiles/x86_energy-static.dir/src/access/sysfs.c.o' failed/home/s9242987/Software/x86_energy/src/access/sysfs.c: In function ‘setup’:
/home/s9242987/Software/x86_energy/src/access/sysfs.c:96:13: error: ‘totalfiles’ undeclared (first use in this function); did you mean ‘total_files’?
         if (totalfiles == 0)
             ^~~~~~~~~~
             total_files
/home/s9242987/Software/x86_energy/src/access/sysfs.c:96:13: note: each undeclared identifier is reported only once for each function it appears in
CMakeFiles/x86_energy-static.dir/build.make:278: recipe for target 'CMakeFiles/x86_energy-static.dir/src/access/sysfs.c.o' failed
```

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -70,7 +70,7 @@ static int init()
             free(namelist);
         }
         closedir(test);
-        if ( found_index )
+        if ( found_index == total_files )
             return 0;
         else
         {

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -81,7 +81,7 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
     int given_package = index;
     char* name = sysfs_names[counter_type];
     struct dirent** namelist;
-    int n, ret, total_files;
+    int n, ret, total_files, read_items=0;
     char file_name_buffer[2048];
     FILE* final_fp = NULL;
     long long int final_max = -1;
@@ -93,19 +93,13 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
 
         n = total_files = scandir(RAPL_PATH, &namelist, NULL, alphasort);
 
-        if (total_files == 0)
-        {
-            X86_ENERGY_SET_ERROR("No valid devices in %s", RAPL_PATH);
-            return NULL;
-        }
-
         while (n--)
         {
             int package;
             int dummy;
             // first we go through all
             // /sys/class/powercap/intel-rapl\:<N> and /sys/class/powercap/intel-rapl\:<N>:<M>
-            int read_items = sscanf(namelist[n]->d_name, "intel-rapl:%d:%d", &package, &dummy);
+            read_items = sscanf(namelist[n]->d_name, "intel-rapl:%d:%d", &package, &dummy);
             if (read_items <= 0)
                 continue;
             // now we verify that we are using the correct package in
@@ -185,9 +179,18 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
 
     if (final_fp == NULL)
     {
-        X86_ENERGY_SET_ERROR("could not get a file pointer to \"%s\"", file_name_buffer);
-        return NULL;
+        if ( read_items <= 0 )
+        {
+            X86_ENERGY_SET_ERROR("No valid files found in " RAPL_PATH);
+            return NULL;
+        }
+        else
+        {
+            X86_ENERGY_SET_ERROR("could not get a file pointer to \"%s\"", file_name_buffer);
+            return NULL;
+        }
     }
+
     if (final_max == -1)
     {
         fclose(final_fp);

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -93,7 +93,7 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
 
         n = total_files = scandir(RAPL_PATH, &namelist, NULL, alphasort);
 
-        if (totalfiles == 0)
+        if (total_files == 0)
         {
             X86_ENERGY_SET_ERROR("No valid devices in %s", RAPL_PATH);
             return NULL;

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -92,6 +92,13 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
         closedir(test);
 
         n = total_files = scandir(RAPL_PATH, &namelist, NULL, alphasort);
+
+        if (totalfiles == 0)
+        {
+            X86_ENERGY_SET_ERROR("No valid devices in %s", RAPL_PATH);
+            return NULL;
+        }
+
         while (n--)
         {
             int package;
@@ -173,12 +180,6 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
     else
     {
         X86_ENERGY_SET_ERROR("can't open RAPL_PATH (%s)", RAPL_PATH);
-        return NULL;
-    }
-
-    if (read_items <= 0)
-    {
-        X86_ENERGY_SET_ERROR("No valid devices in %s", RAPL_PATH);
         return NULL;
     }
 

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -64,7 +64,7 @@ static int init()
             if ( i < total_files )
                 found_index = i;
             else
-                found_index = total_files
+                found_index = total_files;
             for (i = 0; i < found_index; i++)
                 free(namelist[i]);
             free(namelist);

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -65,12 +65,12 @@ static int init()
                 found_index = i;
             else
                 found_index = total_files;
-            for (i = 0; i < found_index; i++)
+            for (i = 0; i < total_files; i++)
                 free(namelist[i]);
             free(namelist);
         }
         closedir(test);
-        if ( found_index == total_files )
+        if ( found_index < total_files )
             return 0;
         else
         {

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -46,8 +46,37 @@ static int init()
     DIR* test = opendir(RAPL_PATH);
     if (test != NULL)
     {
+        int i, total_files, found_index=0;
+        int package;
+        int dummy;
+        struct dirent** namelist;
+        total_files = scandir(RAPL_PATH, &namelist, NULL, alphasort);
+        if ( total_files > 0 )
+        {
+            for ( i = 0; i < total_files ; i++ )
+            {
+                read_items = sscanf(namelist[i]->d_name, "intel-rapl:%d:%d", &package, &dummy);
+                if (read_items > 0)
+                {
+                    break;
+                }
+            }
+            if ( i < total_files )
+                found_index = i;
+            else
+                found_index = total_files
+            for (i = 0; i < found_index; i++)
+                free(namelist[i]);
+            free(namelist);
+        }
         closedir(test);
-        return 0;
+        if ( found_index )
+            return 0;
+        else
+        {
+            X86_ENERGY_SET_ERROR("No valid entries in RAPL_PATH (", RAPL_PATH," )");
+            return 1;
+        }
     }
     X86_ENERGY_SET_ERROR("RAPL_PATH (%s) can not be read", RAPL_PATH);
     return 1;

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -55,7 +55,7 @@ static int init()
         {
             for ( i = 0; i < total_files ; i++ )
             {
-                read_items = sscanf(namelist[i]->d_name, "intel-rapl:%d:%d", &package, &dummy);
+                int read_items = sscanf(namelist[i]->d_name, "intel-rapl:%d:%d", &package, &dummy);
                 if (read_items > 0)
                 {
                     break;

--- a/src/access/sysfs.c
+++ b/src/access/sysfs.c
@@ -176,6 +176,12 @@ static x86_energy_single_counter_t setup(enum x86_energy_counter counter_type, s
         return NULL;
     }
 
+    if (read_items <= 0)
+    {
+        X86_ENERGY_SET_ERROR("No valid devices in %s", RAPL_PATH);
+        return NULL;
+    }
+
     if (final_fp == NULL)
     {
         X86_ENERGY_SET_ERROR("could not get a file pointer to \"%s\"", file_name_buffer);

